XWOptUnity/CraftFactory.cs

```diff
@@ -0,0 +1,349 @@
+        public GameObject HardpointBase { get; set; }
+
+        /// <summary>
+        /// Targeting point objects are cloned from this GameObject
+        /// </summary>
+        public GameObject TargetPointBase { get; set; }
+
+        /// <summary>
+        /// If set, parts with the same target ID are grouped as children of this object.
+        /// </summary>
+        public GameObject TargetingGroupBase { get; set; }
+
+        /// <summary>
+        /// Callback for game specific setup of part objects after instantiation.
+        /// </summary>
+        public ProcessPartHandler ProcessPart { get; set; }
+
+        /// <summary>
+        /// Callback for game specific setup of hardpoint objects after instantiation.
+        /// </summary>
+        public ProcessHardpointHandler ProcessHardpoint { get; set; }
+
+        /// <summary>
+        /// Callback for game specific setup of targeting groups.
+        /// </summary>
+        public ProcessTargetGroupHandler ProcessTargetGroup { get; set; }
+
+        /// <summary>
+        /// The shader to use on the materials.  Default is Unity "Standard" shader.
+        /// </summary>
+        public Shader PartShader { get; set; } = Shader.Find("Standard");
+
+        /// <summary>
+        /// Overall size of the craft.  Used for LOD cutover.
+        /// </summary>
+        public float Size { get; private set; }
+
+        /// <summary>
+        /// Generate emissive textures using low light level pallets.
+        /// This allows things like engines and windows to "glow in the dark."
+        /// Note this will cause the lowest possible darkness to be clamped at the original game's lowest level.
+        /// </summary>
+        public bool MakeEmissiveTexture { get; set; } = true;
+
+        /// <summary>
+        /// Exponent for darkening emissive textures.
+        /// Set high enough to make the dark parts nearly black while leaving emissive parts bright.
+        /// </summary>
+        public float EmissiveExponent { get; set; } = 2f;
+
+        public OptFile<Vector2, Vector3> Opt { get; private set; }
+
+        internal Dictionary<string, Material> materials;
+        List<PartFactory> nonTargetGroupedParts = new List<PartFactory>();
+        Dictionary<DistinctTargetGroupTuple, TargetGroupFactory> targetGroups = new Dictionary<DistinctTargetGroupTuple, TargetGroupFactory>();
+
+        // XvT engine -> Unity engine
+        // unity: forward is +z, right is +x,    up is +y
+        // XvT:   forward is -y, right is +x(?), up is +z
+        static readonly Matrix4x4 CoordinateConverter = new Matrix4x4(
+            new Vector4(1, 0, 0, 0),
+            new Vector4(0, 0, -1, 0),
+            new Vector4(0, 1, 0, 0),
+            new Vector4(0, 0, 0, 1)
+        ) * Matrix4x4.Scale(new Vector3(ScaleFactor, ScaleFactor, ScaleFactor));
+
+        // Size conversion between OPT coordinates and Unity (inch -> meter)
+        public const float ScaleFactor = 1f / 39.37f;
```

Comment:
XWA and XvT use a scale factor of `0.0244140625f` ( `1600.0 * 1.52587890625E-05` ).